provider/kubernetes/kubernetes.go

```diff
@@ -315,6 +309,48 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 	return &templateObjects, nil
 }
 
+func getEntrypoints(i *v1beta1.Ingress) []string {
+	entrypointsAnnotation, ok := i.Annotations[types.LabelFrontendEntryPoints]
+	if ok {
+		return strings.Split(entrypointsAnnotation, ",")
+	}
+	return nil
+}
+
+func getRulePassHostHeader(i *v1beta1.Ingress, p *Provider) bool {
+	passHostHeader := p.getPassHostHeader()
+
+	passHostHeaderAnnotation, ok := i.Annotations[types.LabelFrontendPassHostHeader]
+	switch {
+	case !ok:
+		// No op.
+	case passHostHeaderAnnotation == "false":
+		passHostHeader = false
+	case passHostHeaderAnnotation == "true":
+		passHostHeader = true
+	default:
+		log.Warnf("Unknown value '%s' for %s, falling back to %s", passHostHeaderAnnotation, types.LabelFrontendPassHostHeader, passHostHeader)
+	}
+	return passHostHeader
+}
+
+func getRulePassTLSCert(i *v1beta1.Ingress, p *Provider) bool {
+	passTLSCert := p.getPassTLSCert()
+
+	passTLSCertAnnotation, ok := i.Annotations[types.LabelFrontendPassTLSCert]
+	switch {
+	case !ok:
+		// No op.
+	case passTLSCertAnnotation == "false":
+		passTLSCert = false
+	case passTLSCertAnnotation == "true":
+		passTLSCert = true
+	default:
+		log.Warnf("Unknown value '%s' for %s, falling back to %s", passTLSCertAnnotation, types.LabelFrontendPassTLSCert, passTLSCert)
+	}
+	return passTLSCert
```

Comment:
Is the parsing logic identical to what we have in the function just above? If so, can we extract a little helper and use it from both implementation?

Reply:
Done.